widget/edit.controller.js

```diff
@@ -34,12 +70,43 @@
           $scope.params.fieldsArray = entity.getFormFieldsArray();
         });
       }
-      
+
+      function onChangeModuleType() {
+        delete $scope.config.query;
+        delete $scope.config.customModuleField;
+        delete $scope.config.customModule;
+      }
+
       function init() {
+        $scope.tagsField  = 
+          new Field({
+          name: 'Tags',
+          writeable: true,
+          title: 'Tags',
+          formType: 'tags',
+          dataSource: {
+              model: 'recordTags'
+          }
+        }),
+      appModulesService.load(true).then(function (modules) {
+        $scope.modules = modules;
+
+        //Create a list of modules with atleast one JSON field
+        modules.forEach((module, index) =>{
+          var moduleMetaData = modelMetadatasService.getMetadataByModuleType(module.type);
+          for(let fieldIndex =0; fieldIndex < moduleMetaData.attributes.length; fieldIndex++){
+            //Check If JSON field is present in the module
+            if(moduleMetaData.attributes[fieldIndex].type === "object"){
```

Comment:
Also check if type object is not empty

Reply:
Object does not have data, it just represents if there is a fields with type "json" in the module
Not fetching the value, rather the metadata of the moduke